internal/eval/eval.go

```diff
@@ -587,6 +587,71 @@ func evalJSONARRLEN(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(arrlenList, false)
 }
 
+
+// evalJSONOBJLEN return the number of keys in the JSON object at path in key.
+// Returns an array of integer replies, an integer for each matching value,
+// which is the json objects length, or nil, if the matching value is not a json.
+// Returns encoded error if the key doesn't exist or key is expired or the matching value is not an array.
+// Returns encoded error response if incorrect number of arguments
+func evalJSONOBJLEN(args []string, store *dstore.Store) []byte {
+	if len(args) < 1 {
+		return diceerrors.NewErrArity("JSON.OBJLEN")
+	}
+
+	key := args[0]
+
+	// Retrieve the object from the database
+	obj := store.Get(key)
+	if obj == nil {
+		return clientio.RespNIL
+	}
+
+	// check if the object is json
+	errWithMessage := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeJSON, object.ObjEncodingJSON)
+	if errWithMessage != nil {
+		return errWithMessage
+	}
+
+	// get the value & check for marsheling error
+	jsonData := obj.Value
+	_,err := sonic.Marshal(jsonData)
+	if err != nil {
+		return diceerrors.NewErrWithMessage("Existing key has wrong Dice type")
+	}
+
+
+	if len(args) == 1 {
+		// check if the value is of json type
+		if utils.GetJSONFieldType(jsonData) == utils.ObjectType {
+			return clientio.Encode(len(jsonData.(map[string]interface{})), false)
```

Comment:
Let's not do type casting on the fly, this can lead to server panic if the cast fails.
Instead use
```go
casted, ok := jsonData.(map[string] interface {})
```
And check the ok value to be true

Reply:
done . [commit]( https://github.com/DiceDB/dice/pull/593/commits/54139be1a471092a6f00e0cda24b0ac225da38d3)